solution/ui/regulations/js/src/components/BlockingModal.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+    <div
+        ref="modalElement"
+        class="blocking-modal"
+        :class="activeClass"
+        role="dialog"
+        aria-modal="true"
+        :aria-label="modalTitle"
+    >
+        <div class="blocking-modal-content">
+            <div class="control-row">
+                <button ref="closeBtn" class="close-modal" @click="closeModal">
+                    <span class="close-btn-label">Close</span>
+                    <svg
+                        width="19"
+                        height="19"
+                        viewBox="0 0 19 19"
+                        fill="none"
+                        xmlns="http://www.w3.org/2000/svg"
+                    >
+                        <path
+                            fill-rule="evenodd"
+                            clip-rule="evenodd"
+                            d="M1.85355 1.14645L9.14589 8.43889L16.4393 1.14645C16.6346 0.951184 16.9512 0.951184 17.1464 1.14645C17.3417 1.34171 17.3417 1.65829 17.1464 1.85355L9.85289 9.14589L17.1464 16.4393C17.3417 16.6346 17.3417 16.9512 17.1464 17.1464C16.9512 17.3417 16.6346 17.3417 16.4393 17.1464L9.14589 9.85289L1.85355 17.1464C1.65829 17.3417 1.34171 17.3417 1.14645 17.1464C0.951184 16.9512 0.951184 16.6346 1.14645 16.4393L8.43889 9.14589L1.14645 1.85355C0.951184 1.65829 0.951184 1.34171 1.14645 1.14645C1.34171 0.951184 1.65829 0.951184 1.85355 1.14645Z"
+                            fill="white"
+                            stroke="white"
+                            stroke-width="2"
+                        />
+                    </svg>
+                </button>
+            </div>
+            <iframe
+                width="100%"
+                height="100%"
+                src="https://www.cms.gov"
+                frameborder="0"
+            >
+            </iframe>
+        </div>
+    </div>
+</template>
+
+<script>
+import { EventCodes, trapFocus } from "../../utils";
+
+const POSSIBLE_LOCATIONS = {
+    django: "legacy django template",
+    vite: "vite single page app",
+};
+
+export default {
+    name: "BlockingModal",
+
+    components: {},
+
+    props: {
+        whereUsed: {
+            validator: (value) =>
+                Object.keys(POSSIBLE_LOCATIONS).includes(value),
+            type: String,
+            default: "django",
+        },
+    },
+
+    beforeCreate() {},
+
+    created() {},
+
+    beforeMount() {},
+
+    mounted() {
+        if (this.whereUsed === "vite") {
+            const templateTopContainer = document.getElementById("vue-app");
+            templateTopContainer.insertBefore(
+                this.$el,
+                templateTopContainer.firstChild
+            );
```

Comment:
You can insert Vue components elsewhere in the DOM if needed.  

https://forum.vuejs.org/t/components-with-dom-outside-of-el/2576/2

We need to insert the modal alongside the element that contains the rest of the site content so we can explicitly tell screen readers/assistive technology to ignore the rest of the site content for a11y compliance when the modal is active. 

See `watch` lifecycle method below for how we add/remove `aria-hidden` to all site elements that aren't the blocking modal.